ui/app/js/services/rules_service.js

```diff
@@ -19,6 +19,13 @@ angular.module("app").factory('Rules', function($http, ScheduledChanges) {
     },
     updateRule: function(id, data, csrf_token) {
       data.csrf_token = csrf_token;
+      for (var i in data) {
+        if (data.hasOwnProperty(i)) {
+          if (data[i] === "") {
```

Comment:
What will happen if data[i] is not a string? As far as I know we don't have any nullable fields that aren't strings, but it would be good to make sure this doesn't cause errors integers or objects. The approach looks fine otherwise though.